File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/note/impl/InMemoryNoteRepository.java
package at.tugraz.tc.cyfile.note.impl;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;
import java.util.UUID;

import at.tugraz.tc.cyfile.domain.Note;
import at.tugraz.tc.cyfile.note.NoteRepository;

/**
 * In memory implementation of {@link NoteRepository}
 * Created by cobri on 3/21/2018.
 */

public class InMemoryNoteRepository implements NoteRepository {

    private final Set<Note> noteSet;

    public InMemoryNoteRepository() {
        this.noteSet = new HashSet<>();
    }

    public InMemoryNoteRepository(Set<Note> noteSet) {
        this.noteSet = noteSet;
    }

    @Override
    public List<Note> findAll() {
        return new ArrayList<>(noteSet);
    }

    @Override
    public Note findOne(String id) {
        for (Note note :
                noteSet) {
            if (note.getId().equals(id)) {
                return note;
            }
        }
        return null;
    }

    @Override
    public Note save(Note note) {
        if (note == null) {
            throw new IllegalArgumentException("Cannot be null");
        }

        for (Note n : noteSet) {
            if (n.getId().equals(note.getId())) {
                n.setTitle(note.getTitle());
                n.setContent(note.getContent());
                return n;
            }
        }
        //set id for null
        if (note.getId() == null) {
            note.setId(UUID.randomUUID().toString());
        }
        noteSet.add(note);
        return note;

    }

    @Override
    public void delete(String id) {
        for (Note note :
                noteSet) {
            if (note.getId().equals(id)) {
                noteSet.remove(note);
                return;
            }
        }
    }
}

File: CyFile-Android/app/src/main/java/at/tugraz/tc/cyfile/note/impl/InMemoryNoteRepository.kt
package at.tugraz.tc.cyfile.note.impl

import at.tugraz.tc.cyfile.domain.Note
import at.tugraz.tc.cyfile.note.NoteRepository
import java.util.*

/**
 * In memory implementation of [NoteRepository]
 * Created by cobri on 3/21/2018.
 */

class InMemoryNoteRepository(private val noteSet: MutableSet<Note>) : NoteRepository {
    override fun findAll(): List<Note> {
        return ArrayList(noteSet)
    }

    override fun findOne(id: String): Note? {
        return noteSet.first { x -> x.id == id }
    }

    override fun save(note: Note): Note {
        for (n in noteSet) {
            if (n.id == note.id) {
                n.title = note.title
                n.content = note.content
                n.dateTimeCreated = note.dateTimeCreated
                n.dateTimeModified = note.dateTimeModified
                return n
            }
        }


        //set id for null
        note.id = UUID.randomUUID().toString()

        noteSet.add(note)
        return note

    }

    override fun delete(id: String) {
        noteSet.filter { x -> x.id == id }
                .forEach { x -> noteSet.remove(x) }

    }

    override fun initialize() {
        //nothing to do
    }
}
